azurerm/data_source_express_route_circuit_test.go

```diff
@@ -0,0 +1,83 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccDataSourceAzureRMExpressRoute_basic(t *testing.T) {
+	dataSourceName := "data.azurerm_express_route_circuit.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMExpressRouteCircuitDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMExpressRoute_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "service_provider_properties.0.service_provider_name", "Equinix Test"),
+					resource.TestCheckResourceAttr(dataSourceName, "service_provider_properties.0.peering_location", "Silicon Valley Test"),
+					resource.TestCheckResourceAttr(dataSourceName, "service_provider_properties.0.bandwidth_in_mbps", "50"),
+					resource.TestCheckResourceAttr(dataSourceName, "sku.0.tier", "Standard"),
+					resource.TestCheckResourceAttr(dataSourceName, "sku.0.family", "MeteredData"),
+					resource.TestCheckResourceAttr(dataSourceName, "service_provider_provisioning_state", "NotProvisioned"),
+				),
+			},
+		},
+	})
+}
+
+/*func TestAccDataSourceAzureRMExpressRoute_complete(t *testing.T) {
+	dataSourceName := "data.azurerm_express_route_circuit.test"
+	ri := tf.AccRandTimeInt()
+
+}*/
+
+func testAccDataSourceAzureRMExpressRoute_basic(rInt int, location string) string {
+	config := testAccAzureRMExpressRouteCircuit_MeteredBasic(rInt, location)
```

Comment:
rather than re-defining this we should be able to call [the `testAccAzureRMExpressRouteCircuit_basicMeteredConfig` test method within the Express Route resource](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_express_route_circuit_test.go#L332)

Reply:
That's what I meant to do. But, the subscription I used to test did not support the Express Route configuration that is hard-coded in the test you referenced. But, I checked another subscription and it looks like it does support those configurations. I'll make the change. thanks!